django/db/models/fields/__init__.py

```diff
@@ -737,7 +737,8 @@ def contribute_to_class(self, cls, name, private_only=False):
             # Don't override classmethods with the descriptor. This means that
             # if you have a classmethod and a field with the same name, then
             # such fields can't be deferred (we don't have a check for this).
-            if not getattr(cls, self.attname, None):
+            existing_attr = getattr(cls, self.attname, None)
+            if existing_attr is None or not callable(existing_attr):
```

Comment:
I guess I'd pull this into a helper, since you're going to have to add `@property` decorators (at least) as well.

Reply:
Hey Carlton,
Thanks for following up on this.
Will update soon.